driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoIterable.java

```diff
@@ -55,18 +55,20 @@ public T first() {
 
     @Override
     public void forEach(final Consumer<? super T> action) {
-        SyncSubscriber<T> syncSubscriber = new SyncSubscriber<>();
-        wrapped.subscribe(syncSubscriber);
-        for (T cur : syncSubscriber.all()) {
-            action.accept(cur);
+        try (MongoCursor<T> cursor = cursor()) {
```

Comment:
Since we have a working cursor just use it here and remove the `all` method from the subscriber.